--- a/lib/_emerge/depgraph.py	2026-09-25 16:26:00.213077028 -0500
+++ b/lib/_emerge/depgraph.py	2026-09-25 16:27:34.505108483 -0500
@@ -12185,7 +12185,7 @@
     debug = "--debug" in myopts
     mydepgraph = None
     nodeps = "--nodeps" in myopts
-    max_retries = 0 if nodeps else myopts.get("--backtrack", 20)
+    max_retries = 0 if nodeps else myopts.get("--backtrack", 6)
     max_depth = max(1, (max_retries + 1) // 2)
     allow_backtracking = max_retries > 0
     backtracker = Backtracker(max_depth)
--- a/man/emerge.1	2026-09-25 16:27:59.101506111 -0500
+++ b/man/emerge.1	2026-09-25 16:28:40.618019159 -0500
@@ -452,7 +452,7 @@
 .BR \-\-backtrack=COUNT
 Specifies an integer number of times to backtrack if
 dependency calculation fails due to a conflict or an
-unsatisfied dependency (default: \'20\').
+unsatisfied dependency (default in Funtoo Linux: \'6\').
 .TP
 .BR "\-\-binpkg\-changed\-deps [ y | n ]"
 Tells emerge to ignore binary packages for which the corresponding
